test/test_standard_bookmark.py

```diff
@@ -0,0 +1,196 @@
+
+        ##########################################################################
+        # Test By Stream
+        ##########################################################################
+
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                # expected values
+                expected_replication_method = expected_replication_methods[stream]
+
+                # collect information for assertions from syncs 1 & 2 base on expected values
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+                first_sync_messages = [record.get('data') for record in
+                                       first_sync_records.get(
+                                           stream, {}).get('messages', [])
+                                       if record.get('action') == 'upsert']
+                second_sync_messages = [record.get('data') for record in
+                                        second_sync_records.get(
+                                            stream, {}).get('messages', [])
+                                        if record.get('action') == 'upsert']
+                first_bookmark_key_value = first_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                second_bookmark_key_value = second_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                
+
+                if expected_replication_method == self.INCREMENTAL:
+
+                    # collect information specific to incremental streams from syncs 1 & 2
+                    replication_key = next(
+                        iter(expected_replication_keys[stream]))
+                    first_bookmark_value = first_bookmark_key_value.get(replication_key)
+                    second_bookmark_value = second_bookmark_key_value.get(replication_key)
+                    first_bookmark_value_utc = self.convert_state_to_utc(
+                        first_bookmark_value)
+                    second_bookmark_value_utc = self.convert_state_to_utc(
+                        second_bookmark_value)
+
+                    
+                    simulated_bookmark_value = self.convert_state_to_utc(new_states['bookmarks'][stream][replication_key])
+                    
+                    # Verify the first sync sets a bookmark of the expected form
+                    self.assertIsNotNone(first_bookmark_key_value)
+                    self.assertIsNotNone(first_bookmark_value)
+
+                    # Verify the second sync sets a bookmark of the expected form
+                    self.assertIsNotNone(second_bookmark_key_value)
+                    self.assertIsNotNone(second_bookmark_value)
+
+                    # Verify the second sync bookmark is Equal to the first sync bookmark
+                    # assumes no changes to data during test
+                    if not stream == "users":
```

Comment:
Streams should not be skipped like this, there needs to be an explanation and tickets linked if applicable.

Reply:
For `users` stream it stores bookmark as 1 minute less than current time(existing tap behavior) if `updated_at` of last records less than it. So, if there is no data change during second sync then second_bookmark_value will be 1 minute less than current time. That's why, skipping below assertion for users stream. Added comment in test case also.

Reply:
Without knowing the implementation details, I would generally consider this a bug. If this implementation is intentional there should be a separate assertion that covers the expected behavior of just the `users` stream.

Reply:
Added expected separate assertion for user stream.